sdk/storage/storage-blob/src/BlobBatchClient.ts

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as Models from "./generated/src/models";
+import { ParsedBatchResponse } from "./BatchResponse";
+import { BatchResponseParser } from "./BatchResponseParser";
+import { utf8ByteLength } from "./BatchUtils";
+import { BlobBatch } from "./BlobBatch";
+import { CommonOptions, BlobDeleteOptions, BlobClient, BlobSetTierOptions } from "./internal";
+import { AbortSignalLike } from "@azure/abort-controller";
+import { CanonicalCode } from "@azure/core-tracing";
+import { createSpan } from "./utils/tracing";
+import { HttpResponse, TokenCredential } from "@azure/core-http";
+import { Service } from "./generated/src/operations";
+import { SharedKeyCredential } from "./credentials/SharedKeyCredential";
+import { AnonymousCredential } from "./credentials/AnonymousCredential";
+
+/**
+ * Options to configure the Service - Submit Batch Optional Params.
+ *
+ * @export
+ * @interface ServiceSubmitBatchOptionalParams
+ */
+export interface ServiceSubmitBatchOptionalParams
+  extends Models.ServiceSubmitBatchOptionalParams,
+    CommonOptions {
+  /**
+   * An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
+   *
+   * @type {AbortSignalLike}
+   * @memberof ServiceSubmitBatchOptionalParams
+   */
+  abortSignal?: AbortSignalLike;
+}
+
+export declare type ServiceSubmitBatchResponse = ParsedBatchResponse &
```

Comment:
When creating these, how do you decide between using an interface or a type? I see that `ServiceSubmitBatchOptionalParams` is an interface and `ServiceSubmitBatchResponse` is a type

Reply:
I believe this is the syntax for type aliases?

---
In reply to: [336240754](https://github.com/Azure/azure-sdk-for-js/pull/5634#discussion_r336240754) [](ancestors = 336240754)

Reply:
I mean really everything is a type ultimately, an interface is just a specific shape of an object type. 

The syntax for declaring types lets you work with primitive values (e.g. this can be a string or an interface) and has nice syntax for unions and intersections whereas interfaces only have composition.

Reply:
Here's the official word on it with nuance: https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases